src/main/java/com/taxi/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,312 @@
+    private final ConnectionUtil connectionUtil;
+
+    @Autowired
+    public CarDaoImpl(ConnectionUtil connectionUtil) {
+        this.connectionUtil = connectionUtil;
+    }
+
+    @Override
+    public Car create(Car car) {
+        Connection connection = connectionUtil.getConnection();
+        try {
+            connection.setAutoCommit(false);
+            insertCarDataToDb(car, connection);
+            addCarDrivers(car, connection);
+            connection.commit();
+        } catch (SQLException e) {
+            try {
+                connection.rollback();
+            } catch (SQLException ex) {
+                throw new RuntimeException("Rollback is not successful", ex);
+            }
+            throw new DataProcessingException("Can't create car: "
+                    + car, e);
+        } finally {
+            try {
+                connection.setAutoCommit(true);
+                connection.close();
+            } catch (SQLException e) {
+                throw new RuntimeException("Couldn't close the connection", e);
+            }
+        }
+        return car;
+    }
+
+    private Car insertCarDataToDb(Car car, Connection connection) throws SQLException {
+        String queryCreate = "INSERT INTO cars (manufacturer_id, model) VALUES (?, ?)";
+        try (
+                PreparedStatement statement
+                        = connection.prepareStatement(queryCreate,
+                        PreparedStatement.RETURN_GENERATED_KEYS)
+        ) {
+            statement.setLong(1, car.getManufacturer().getId());
+            statement.setString(2, car.getModel());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            resultSet.next();
+            Long insertedId = resultSet.getObject(1, Long.class);
+            car.setId(insertedId);
+        }
+        return car;
+    }
+
+    @Override
+    public Car update(Car car) {
+        Connection connection = connectionUtil.getConnection();
+        try {
+            connection.setAutoCommit(false);
+            modifyCarDataInDb(car, connection);
+            removeCarDrivers(car, connection);
+            addCarDrivers(car, connection);
+            connection.commit();
+        } catch (SQLException e) {
+            try {
+                connection.rollback();
+            } catch (SQLException ex) {
+                throw new RuntimeException("Rollback is not successful", ex);
+            }
+            throw new DataProcessingException("Can't update car: "
+                    + car, e);
+        } finally {
+            try {
+                connection.setAutoCommit(true);
+                connection.close();
+            } catch (SQLException e) {
+                throw new RuntimeException("Couldn't close the connection", e);
+            }
+        }
+        return car;
+    }
+
+    private Car modifyCarDataInDb(Car car, Connection connection) throws SQLException {
```

Comment:
Стараемся придерживаться единой терминологии
```suggestion
    private Car updateCarDataInDb(Car car, Connection connection) throws SQLException {
```

Reply:
переписав create та update